manifests/stage-rdma-device-plugin/0020-k8s-rdma-shared-dev-plugin-ds.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 NVIDIA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: rdma-shared-dp-ds
+  namespace: {{ .RuntimeSpec.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      app: rdma-shared-dp
+  template:
+    metadata:
+      # Mark this pod as a critical add-on; when enabled, the critical add-on scheduler
+      # reserves resources for critical add-on pods so that they can be rescheduled after
+      # a failure.  This annotation works in tandem with the toleration below.
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels:
+        app: rdma-shared-dp
+    spec:
+      hostNetwork: true
+      tolerations:
+      # Allow this pod to be rescheduled while the node is in "critical add-ons only" mode.
+      # This, along with the annotation above marks this pod as a critical add-on.
+      - key: CriticalAddonsOnly
+        operator: Exists
+      containers:
+      - image: {{ .CrSpec.Repository }}/{{ .CrSpec.Image }}:{{ .CrSpec.Version }}
+        name: rdma-shared-dp
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
```

Comment:
why it privileged? because of the socket?

Reply:
Thats how its defined : https://github.com/Mellanox/k8s-rdma-shared-dev-plugin/blob/39f724187de4169c7b9d47ba8c861179c6803724/images/k8s-rdma-shared-dev-plugin-ds.yaml#L31

I did not check if it can work without

Reply:
ok fine